iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.cs

```diff
@@ -1,101 +1,108 @@
-                  {
-                    Console.WriteLine("  {0}: {1}", prop.Key, prop.Value);
-                  }
-                  Console.WriteLine("System properties (set by IoT Hub):");
-                  foreach (var prop in eventData.SystemProperties)
-                  {
-                    Console.WriteLine("  {0}: {1}", prop.Key, prop.Value);
-                  }
-                }
-            }
-        }
-
-        private static async Task Main(string[] args)
-        {
-            Console.WriteLine("IoT Hub Quickstarts - Read device to cloud messages. Ctrl-C to exit.\n");
-
-            // Create an EventHubClient instance to connect to the
-            // IoT Hub Event Hubs-compatible endpoint.
-            var connectionString = new EventHubsConnectionStringBuilder(new Uri(s_eventHubsCompatibleEndpoint), s_eventHubsCompatiblePath, s_iotHubSasKeyName, s_iotHubSasKey);
-            s_eventHubClient = EventHubClient.CreateFromConnectionString(connectionString.ToString());
-
-            // Create a PartitionReciever for each partition on the hub.
-            var runtimeInfo = await s_eventHubClient.GetRuntimeInformationAsync();
-            var d2cPartitions = runtimeInfo.PartitionIds;
-
-            CancellationTokenSource cts = new CancellationTokenSource();
-
-            Console.CancelKeyPress += (s, e) =>
-            {
-                e.Cancel = true;
-                cts.Cancel();
-                Console.WriteLine("Exiting...");
-            };
-
-            var tasks = new List<Task>();
-            foreach (string partition in d2cPartitions)
-            {
-                tasks.Add(ReceiveMessagesFromDeviceAsync(partition, cts.Token));
-            }
-
-            // Wait for all the PartitionReceivers to finsih.
-            Task.WaitAll(tasks.ToArray());
-        }
-    }
-}
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This application uses the Azure Event Hubs Client for .NET
+// For samples see: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/README.md
+// For documentation see: https://docs.microsoft.com/azure/event-hubs/
+
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace read_d2c_messages
+{
+    class ReadDeviceToCloudMessages
+    {
+        // Event Hub-compatible endpoint
+        // az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}
+        private const string EventHubsCompatibleEndpoint = "{your Event Hubs compatible endpoint}";
+
+        // Event Hub-compatible name
+        // az iot hub show --query properties.eventHubEndpoints.events.path --name {your IoT Hub name}
+        private const string EventHubName = "{your Event Hubs compatible name}";

```

Comment:
Not sure why you made this constant **EventHubName** rather than **EventHubsCompatiblePath**. Is there a reason? (I'm not arguing -- just asking.)

Reply:
Yes; the concept in the Event Hubs client across each of the languages in track two  is called `EventHubName`.   The `EntityPath` semantics were largely a holdover from the days of Event Hubs and Service Bus sharing a neutral interface where a queue/topic/subscription/event hub could be specified.

Reply:
Thanks for the context! That doesn't seem worth arguing about -- **EventHubName** it is. :)